### ISSUE TITLE
Build failed: Linking of rule '//tensorflow_serving/core:aspired_versions_manager_benchmark'

### ISSUE THREAD

**Issue by Qululu** [tensorflow/serving#297]
I'm encountering build failures when trying to install a fresh clone of Tensorflow Serving (latest version) (Tensorflow version 0.12 CPU only).

Using Ubuntu 14.04, Bazel 0.4.3, Python 2.7.12

I followed all the steps exactly in [https://tensorflow.github.io/serving/setup](https://tensorflow.github.io/serving/setup).

Here is the error after running `bazel build --verbose_failures tensorflow_serving/...`:

```
WARNING: /home/qululu/.cache/bazel/_bazel_qululu/40bc54f732f66aa5728c0d42450d1ae6/external/inception_model/WORKSPACE:1: Workspace name in /home/qululu/.cache/bazel/_bazel_qululu/40bc54f732f66aa5728c0d42450d1ae6/external/inception_model/WORKSPACE (@inception) does not match the name given in the repository's definition (@inception_model); this will cause a build error in future versions.
INFO: Found 224 targets...
ERROR: /home/qululu/serving/tensorflow_serving/core/BUILD:531:1: Linking of rule '//tensorflow_serving/core:aspired_versions_manager_benchmark' failed: gcc failed: error executing command
  (cd /home/qululu/.cache/bazel/_bazel_qululu/40bc54f732f66aa5728c0d42450d1ae6/execroot/serving && \
  exec env - \
    PATH=/home/qululu/anaconda2/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/home/qululu/bin \
  /usr/bin/gcc -o bazel-out/local-fastbuild/bin/tensorflow_serving/core/aspired_versions_manager_benchmark '-Wl,-rpath,$ORIGIN/../../_solib_k8/' -Lbazel-out/local-fastbuild/bin/_solib_k8 -pthread -Wl,-no-as-needed -B/usr/bin -B/usr/bin -pass-exit-codes '-Wl,--build-id=md5' '-Wl,--hash-style=gnu' -Wl,-S -Wl,@bazel-out/local-fastbuild/bin/tensorflow_serving/core/aspired_versions_manager_benchmark-2.params): com.google.devtools.build.lib.shell.BadExitStatusException: Process exited with status 1.
collect2: error: ld returned 1 exit status
INFO: Elapsed time: 39.136s, Critical Path: 38.76s
```

I also tried the suggested solutions in issues #199 and #227 such as upgrading the gcc version to 4.9 and trying build while accounting for any memory limitations (even though I have 30GB RAM):
`bazel build --local_resources 5000,1.0,1.0 tensorflow_serving/...`

Please help, thanks.

**Comment by kirilg** [comment]
Hmm, I'm not exactly sure what went wrong, the logs don't give enough of a clue.

I see you mentioned you synced to TensorFlow version 0.12. Normally we only test the version of TensorFlow that we pin the repo to, so if you change that, it may not work as you expect. Can you try the latest TensorFlow Serving at head using the TF commit that it's currently pinned to?

**Comment by eth-n** [comment]
I was just having the same error (on core:aspired_versions_manager_benchmark), but on Ubuntu 16.04 with Bazel 0.5.1 and Python 3.5.2. I had used pip3 to install TensorFlow 1.2.0, and then put cuDNN 6.0 in the CUDA include and lib64 directories. Changing cuDNN back to 5.1 (and hence making it compatible with the TensorFlow version I had installed.. whoops) fixed this error; my build just completed and passed `bazel test tensorflow_serving/...`.
